src/platform/packages/shared/kbn-unified-metrics-grid/src/components/metrics_grid_toolbar_wraper.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React, { useMemo } from 'react';
+import { EuiFocusTrap, EuiOverlayMask, useEuiTheme } from '@elastic/eui';
+import type { EuiThemeComputed } from '@elastic/eui';
+import type { CSSObject } from '@emotion/react';
+import { ChartSectionTemplate } from '@kbn/unified-histogram';
+import type { SerializedStyles } from '@emotion/serialize';
+import type { MetricField } from '@kbn/metrics-experience-plugin/common/types';
+import type { ChartSectionProps } from '@kbn/unified-histogram/types';
+import { useMetricsGridState } from '../hooks';
+import { RightSideActions } from './toolbar/right_side_actions/search_input';
+import { useToolbarActions } from './toolbar/hooks/use_toolbar_actions';
+
+const getFullScreenStyles = (euiTheme: EuiThemeComputed): CSSObject => {
+  return {
+    position: 'fixed',
+    top: 0,
+    left: 0,
+    width: '100vw',
+    height: '100vh',
+    background: euiTheme.colors.backgroundBasePlain,
+    display: 'flex',
+    flexDirection: 'column',
+    minHeight: 0,
+    zIndex: euiTheme.levels.modal,
+    overscrollBehavior: 'contain',
+  };
+};
+
+interface FullScreenProps {
+  isFullscreen: boolean;
+  dataTestSubj?: string;
+}
+
+const FullScreenWrapper = ({
+  isFullscreen,
+  dataTestSubj,
+  children,
+}: React.PropsWithChildren<FullScreenProps>) => {
+  const { euiTheme } = useEuiTheme();
+
+  const styles = useMemo(() => getFullScreenStyles(euiTheme), [euiTheme]);
+
+  if (!isFullscreen) return <>{children}</>;
+
+  return (
+    <EuiOverlayMask headerZindexLocation="above">
+      <EuiFocusTrap>
+        <div css={styles} data-test-subj={`${dataTestSubj}FullScreenWrapper`}>
+          {children}
+        </div>
+      </EuiFocusTrap>
+    </EuiOverlayMask>
+  );
+};
+
+export interface MetricsGridToolbarWrapperProps
+  extends Pick<ChartSectionProps, 'requestParams' | 'renderToggleActions'> {
+  indexPattern: string;
+  chartToolbarCss?: SerializedStyles;
+  setDebouncedSearchTerm: (value: string) => void;
+  fields: MetricField[];
+  children?: React.ReactNode;
+}
+
+export const MetricsGridToolbarWrapper = ({
```

Comment:
This wraps more than just the toolbar, right?

```suggestion
export const MetricsGridWrapper = ({
```

Reply:
I renamed it but I would keep it as separate for now as the metric grid is getting loaded, and I can imagine extending it more, so keeping it a separate component can help and also could be reusable in other sections in the future. If we see that it's not the case, we can also refactor it later